### ISSUE TITLE
Generator tree error?

### ISSUE THREAD

**Issue by utahwithak** [Merri/map-generator#1]
https://github.com/Merri/map-generator/blob/master/src/map.js#L382
Should this also include `nodes.bottomRight`

**Comment by Merri** [comment]
No, bottom right node is handled slightly differently in many cases due to flagpole handling.

**Comment by utahwithak** [comment]
Cool. thanks for the response. Another question:
Here is it intentional to be out of bounds at the start and never use the 0th index?
https://github.com/Merri/map-generator/blob/master/src/generator.js#L168

**Comment by Merri** [comment]
`--i` executes first so it loops from last index to 0. It would work as you assume if the loop had `i--`.

**Comment by utahwithak** [comment]
?? I do not think that is at all correct ??
```
var cars = ["BMW", "Volvo", "Saab", "Ford", "Fiat", "Audi"];
var text = "";
var i;
for (i = cars.length; i > 0; --i) {
    text += cars[i] + ", "+i+ "<br>";
}
```
will produce
```
undefined, 6
Audi, 5
Fiat, 4
Ford, 3
Saab, 2
Volvo, 1
```
The increment/decrement still happens after the first iteration. the next if statement that checks for `undefined` handles it. The result being that the first node is ignored.

**Comment by Merri** [comment]
You are right; I didn't have time at work to validate. Usually I avoid writing `--i` as it's rules are occasionally confusing, but in the other hand I also tend to like compactness. Makes me wonder if this particular case happens to work differently in some other C-like language. Anyway, the loop would work as intended by removing `--i` from the conditions entirely and adding it to the first usage within the loop: `index = nodes[--i];`

Since this project is on a hiatus (and I'm lazy) I let the code stand as it is right now. Most of it is quite many years old by now and there are worse issues that need to be resolved (such as replacing `Math.random()` with a seedable solution to be able to generate the exact same map from same input every single time).

But it is great you bother to review it as it is right now :)
